SCM panel doesn't show git conflicts list of files

Issue Type: <b>Bug</b>

This bug happens on version: 1.51.2 and Insiders 1.53.0. Version 1.51.1 works fine.

In a project with git enabled:
1. Do some edits to a file
2. Checkout to a new branch commit, and push to a remote branch
3. Checkout back to master
4. Edit the same file edited before in order to have confilct
5. git pull the previous remote branch
6. SCM button on Side Bar shows the notification number of files but list of changed files remains empty

Console errors:
```
/C:/Program Files/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:1780   ERR [File Watcher (node.js)] Error: ENOENT: no such file or directory, stat 'c:\Projects\geoportal\docker\.vscode'
/C:/Program Files/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:4584 [[object Object]]Should never happen
$onExtensionRuntimeError @ /C:/Program Files/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:4584
/C:/Program Files/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:4584 Error: Should never happen
	at A.getRightResource (c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\dist\main.js:1:72928)
	at A.getResources (c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\dist\main.js:1:72210)
	at C.get resources (c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\dist\main.js:1:64222)
	at C.<anonymous> (c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\dist\main.js:1:23851)
	at C.get leftUri [as leftUri] (c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\dist\main.js:1:64053)
	at A.resolveChangeCommand (c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\dist\main.js:1:71760)
	at A.resolveDefaultCommand (c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\dist\main.js:1:71556)
	at C.get command [as command] (c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\dist\main.js:1:64149)
	at c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:824:888
	at Array.map (<anonymous>)
	at c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:824:670
	at Array.map (<anonymous>)
	at w._takeResourceStateSnapshot (c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:824:643)
	at c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:828:975
	at Set.forEach (<anonymous>)
	at y.eventuallyUpdateResourceStates (c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:828:953)
	at Timeout._onTimeout (c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:8:779)
	at listOnTimeout (internal/timers.js:549:17)
	at processTimers (internal/timers.js:492:7)
$onExtensionRuntimeError @ /C:/Program Files/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:4584
DevTools failed to load SourceMap: Could not load content for https://ticino.blob.core.windows.net/sourcemaps/96b426ef1da0f0a51ce9ef1931cd1fd2322547e4/node_modules/windows-process-tree/lib/index.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

VS Code version: Code - Insiders 1.53.0-insider (96b426ef1da0f0a51ce9ef1931cd1fd2322547e4, 2020-12-14T05:59:27.649Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU E3-1245 v5 @ 3.50GHz (8 x 3504)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.91GB (11.64GB free)|
|Process Argv|--crash-reporter-id f6ad055e-8037-4edc-9ab8-e090701f2d28|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
openlogontheside:30221882
python763:30178808
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
python504:30227505
vswsl492:30208929
wsl2prompt:30219162
vstry244:30230484
pythonvsdeb440:30224570
unusedpromptcf:30219165
folderexplorercf:30219167
openfilemenu:30219168
pythonvsded773:30223139
core-portspanelcf:30233468

```

</details>

<!-- generated by issue reporter -->

--- a/extensions/git/src/repository.ts
+++ b/extensions/git/src/repository.ts
@@ -79,8 +79,8 @@ export class Resource implements SourceControlResourceState {
 		return this.resources[0];
 	}
 
-	get rightUri(): Uri {
-		return this.resources[1] as Uri;
+	get rightUri(): Uri | undefined {
+		return this.resources[1];
 	}
 
 	get command(): Command {

--- a/extensions/git/src/repository.ts
+++ b/extensions/git/src/repository.ts
@@ -80,7 +80,7 @@ export class Resource implements SourceControlResourceState {
 	}
 
 	get rightUri(): Uri {
-		return this.resources[1];
+		return this.resources[1] as Uri;
 	}
 
 	get command(): Command {
@@ -88,7 +88,7 @@ export class Resource implements SourceControlResourceState {
 	}
 
 	@memoize
-	private get resources(): [Uri | undefined, Uri] {
+	private get resources(): [Uri | undefined, Uri | undefined] {
 		return this._commandResolver.getResources(this);
 	}
 

--- a/extensions/git/src/repository.ts
+++ b/extensions/git/src/repository.ts
@@ -613,7 +613,7 @@ class ResourceCommandResolver {
 		}
 	}
 
-	getResources(resource: Resource): [Uri | undefined, Uri] {
+	getResources(resource: Resource): [Uri | undefined, Uri | undefined] {
 		for (const submodule of this.repository.submodules) {
 			if (path.join(this.repository.root, submodule.path) === resource.resourceUri.fsPath) {
 				return [undefined, toGitUri(resource.resourceUri, resource.resourceGroupType === ResourceGroupType.Index ? 'index' : 'wt', { submoduleOf: this.repository.root })];
@@ -641,7 +641,7 @@ class ResourceCommandResolver {
 		return undefined;
 	}
 
-	private getRightResource(resource: Resource): Uri {
+	private getRightResource(resource: Resource): Uri | undefined {
 		switch (resource.type) {
 			case Status.INDEX_MODIFIED:
 			case Status.INDEX_ADDED:
@@ -677,7 +677,7 @@ class ResourceCommandResolver {
 				return resource.resourceUri;
 		}
 
-		throw new Error('Should never happen');
+		return undefined;
 	}
 
 	private getTitle(resource: Resource): string {

--- a/extensions/git/src/test/smoke.test.ts
+++ b/extensions/git/src/test/smoke.test.ts
@@ -139,7 +139,9 @@ suite('git smoke test', function () {
 		cp.execSync('git add .', { cwd });
 		await repository.commit('commit on master');
 
-		cp.execSync('git merge test', { cwd });
+		try {
+			cp.execSync('git merge test', { cwd });
+		} catch (e) {}
 		await new Promise(resolve => setTimeout(resolve, 5e3));
 
 		await commands.executeCommand('workbench.scm.focus');

--- a/extensions/git/src/test/smoke.test.ts
+++ b/extensions/git/src/test/smoke.test.ts
@@ -126,6 +126,8 @@ suite('git smoke test', function () {
 	});
 		
 	test('rename/delete conflict', async function () {
+		await commands.executeCommand('workbench.scm.focus');
+
 		cp.execSync('git branch test', { cwd });
 		cp.execSync('git checkout test', { cwd });
 
@@ -141,10 +143,10 @@ suite('git smoke test', function () {
 
 		try {
 			cp.execSync('git merge test', { cwd });
-		} catch (e) {}
-		await new Promise(resolve => setTimeout(resolve, 5e3));
+		} catch (e) { }
 
-		await commands.executeCommand('workbench.scm.focus');
-		await new Promise(resolve => setTimeout(resolve, 5e3));
+		await new Promise(resolve => {
+			setTimeout(resolve, 5e3);
+		});
 	});
 });

--- a/extensions/git/src/test/smoke.test.ts
+++ b/extensions/git/src/test/smoke.test.ts
@@ -124,4 +124,25 @@ suite('git smoke test', function () {
 		assert.equal(repository.state.workingTreeChanges.length, 0);
 		assert.equal(repository.state.indexChanges.length, 0);
 	});
+		
+	test('rename/delete conflict', async function () {
+		cp.execSync('git branch test', { cwd });
+		cp.execSync('git checkout test', { cwd });
+
+		fs.unlinkSync(file('app.js'));
+		cp.execSync('git add .', { cwd });
+
+		await repository.commit('commit on test');
+		cp.execSync('git checkout master', { cwd });
+
+		fs.renameSync(file('app.js'), file('rename.js'));
+		cp.execSync('git add .', { cwd });
+		await repository.commit('commit on master');
+
+		cp.execSync('git merge test', { cwd });
+		await new Promise(resolve => setTimeout(resolve, 5e3));
+
+		await commands.executeCommand('workbench.scm.focus');
+		await new Promise(resolve => setTimeout(resolve, 5e3));
+	});
 });

--- a/extensions/git/src/test/smoke.test.ts
+++ b/extensions/git/src/test/smoke.test.ts
@@ -126,8 +126,6 @@ suite('git smoke test', function () {
 	});
 		
 	test('rename/delete conflict', async function () {
-		await commands.executeCommand('workbench.scm.focus');
-
 		cp.execSync('git branch test', { cwd });
 		cp.execSync('git checkout test', { cwd });
 
@@ -144,6 +142,10 @@ suite('git smoke test', function () {
 		try {
 			cp.execSync('git merge test', { cwd });
 		} catch (e) { }
+		
+		setTimeout(() => {
+			commands.executeCommand('workbench.scm.focus');
+		}, 2e3);
 
 		await new Promise(resolve => {
 			setTimeout(resolve, 5e3);

